src/main/java/com/fasterxml/jackson/datatype/jsr310/ser/key/Jsr310NullKeySerializer.java

```diff
@@ -0,0 +1,33 @@
+package com.fasterxml.jackson.datatype.jsr310.ser.key;
+
+import java.io.IOException;
+import java.util.Map;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+
+/**
+ * This class is to be used in case {@code null} keys are needed to be serialized in a {@link Map} with Java 8 temporal keys. By default the
+ * {@code null} key is not supported by jackson, the serializer needs to be registered manually.
+ *
+ * @author Zoltan Kiss
+ * @since 2.6
```

Comment:
Are null keys not supported by main `jackson-databind`?

How about I'll merge, and let you verify things are correct (minus possible merging if I need to do it).